server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1362,6 +1372,15 @@ public GatedCloseable<IndexCommit> acquireSafeIndexCommit() throws EngineExcepti
         }
     }
 
+    public ReplicationCheckpoint getLatestReplicationCheckpoint() {
```

Comment:
lets add javadoc for the public methods please.

Reply:
Sure. Added in new commit